app/src/main/java/com/example/unscramble/ui/GameScreen.kt

```diff
@@ -0,0 +1,261 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+
+@Composable
+fun GameScreen(
+    modifier: Modifier = Modifier,
+    gameViewModel: GameViewModel = viewModel()
+) {
+    val gameUiState by gameViewModel.uiState.collectAsState()
+
+    Column(
+        modifier = modifier
+            .verticalScroll(rememberScrollState())
+            .padding(16.dp),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+
+        Text(
+            text = stringResource(R.string.app_name),
+            style = typography.titleLarge,
+        )
+
+        GameLayout(
+            onUserGuessChanged = { gameViewModel.updateUserGuess(it) },
+            wordCount = gameUiState.currentWordCount,
+            userGuess = gameViewModel.userGuess,
+            onKeyboardDone = { gameViewModel.checkUserGuess() },
+            currentScrambledWord = gameUiState.currentScrambledWord,
+            isGuessWrong = gameUiState.isGuessedWordWrong
+        )
+        Column(
+            modifier = modifier
+                .fillMaxWidth()
+                .padding(16.dp),
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+
+            Button(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .padding(top = 16.dp),
+                onClick = { gameViewModel.checkUserGuess() }
+            ) {
+                Text(
+                    text = stringResource(R.string.submit),
+                    fontSize = 16.sp
+                )
+            }
+
+            OutlinedButton(
+                onClick = { gameViewModel.skipWord() },
+                modifier = Modifier.fillMaxWidth()
+            ) {
+                Text(
+                    text = stringResource(R.string.skip),
+                    fontSize = 16.sp
+                )
+            }
+        }
+
+        GameStatus(score = gameUiState.score)
+
+        if (gameUiState.isGameOver) {
+            FinalScoreDialog(
+                score = gameUiState.score,
+                onPlayAgain = { gameViewModel.resetGame() }
+            )
+        }
+    }
+}
+
+@Composable
+fun GameStatus(score: Int, modifier: Modifier = Modifier) {
+    Card(
+        modifier = modifier.padding(20.dp)
+    ) {
+        Text(
+            text = stringResource(R.string.score, score),
+            fontSize = 28.sp,
+            modifier = modifier.padding(8.dp),
+            fontWeight = FontWeight.Bold
+        )
+
+    }
+}
+
+@Composable
+fun GameLayout(
+    currentScrambledWord: String,
+    wordCount: Int,
+    isGuessWrong: Boolean,
+    userGuess: String,
+    onUserGuessChanged: (String) -> Unit,
+    onKeyboardDone: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Card(
+        modifier = modifier
+            .fillMaxWidth()
```

Comment:
have the parent composable pass in the modifier with fillMaxWidth(), wrapContentHeight(), and padding()